flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1788,6 +1800,11 @@ private StateBackend loadStateBackend(ReadableConfig configuration, ClassLoader
                 addSource(monitoringFunction, sourceName, null, boundedness)
                         .transform("Split Reader: " + sourceName, typeInfo, factory);
 
+        if (getParallelism() == ExecutionConfig.PARALLELISM_DEFAULT
+                && getConfig().isDynamicGraph()) {
```

Comment:
I think we can introduce a boolean variable to reuse the `getParallelism() == ExecutionConfig.PARALLELISM_DEFAULT && getConfig().isDynamicGraph()` logic.

Reply:
agree, it's more concise.